EIPS/eip-6981.md

```diff
@@ -0,0 +1,425 @@
+
+### Account Registry and Account Implementation Coupling
+
+Since Account Instances are deployed as [ERC-1167](./eip-1167.md) proxies, the account implementation address affects the addresses of accounts deployed from a given Account Registry. Requiring that registry instances be linked to a single, immutable account implementation ensures consistency between a user's salt and linked address on a given Account Registry Instance.
+
+This also allows services to gain the the trust of users by deploying their registries with a reference to a trusted account implementation address.
+
+Furthermore, account implementations can be designed as upgradeable, so users are not necessarily bound to the implementation specified by the Account Registry Instance used to create their account.
+
+## Reference Implementation
+
+The following is an example of an Account Registry Factory which can be used by external service providers to deploy their own Account Registry Instance.
+
+### Account Registry Factory
+
+```solidity
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.13;
+
+/// @author: manifold.xyz
+
+import {Create2} from "openzeppelin/utils/Create2.sol";
+
+import {Address} from "../../lib/Address.sol";
+import {ERC1167ProxyBytecode} from "../../lib/ERC1167ProxyBytecode.sol";
+import {IAccountRegistryFactory} from "./IAccountRegistryFactory.sol";
+
+contract AccountRegistryFactory is IAccountRegistryFactory {
+    using Address for address;
+
+    error InitializationFailed();
+
+    address private immutable registryImplementation = 0x076B08EDE2B28fab0c1886F029cD6d02C8fF0E94;
+
+    function createRegistry(address implementation, uint96 index) external returns (address) {
+        bytes32 salt = _getSalt(msg.sender, index);
+        bytes memory code = ERC1167ProxyBytecode.createCode(registryImplementation);
+        address _registry = Create2.computeAddress(salt, keccak256(code));
+
+        if (_registry.isDeployed()) return _registry;
+
+        _registry = Create2.deploy(0, salt, code);
+
+        (bool success, ) = _registry.call(
+            abi.encodeWithSignature("initialize(address,address)", implementation, msg.sender)
+        );
+        if (!success) revert InitializationFailed();
+
+        emit AccountRegistryCreated(_registry, implementation, index);
+
+        return _registry;
+    }
+
+    function registry(address deployer, uint96 index) external view override returns (address) {
+        bytes32 salt = _getSalt(deployer, index);
+        bytes memory code = ERC1167ProxyBytecode.createCode(registryImplementation);
+        return Create2.computeAddress(salt, keccak256(code));
+    }
+
+    function _getSalt(address deployer, uint96 index) private pure returns (bytes32) {
+        return bytes32(abi.encodePacked(deployer, index));
+    }
+}
+```
+
+### Account Registry
+
+```solidity
+// SPDX-License-Identifier: MIT
```

Comment:
If the solidity is included inline in the proposal, it must have a license of CC0-1.0

Reply:
Updated!